api/src/services/__tests__/configService.ts

```diff
@@ -0,0 +1,57 @@
+import { existsSync, readFileSync } from 'fs';
+
+import { clearConfig, findConfig } from '../configService';
+
+jest.mock('fs');
+const mockExistsSync = jest.mocked(existsSync);
+const mockReadFileSync = jest.mocked(readFileSync);
+
+describe('configService', () => {
+  describe('findConfig', () => {
+    it('should read a secrets file if one exists with the key name', () => {
+      const key = 'TEST_KEY';
+      const secretValue = 'secret value';
+      mockExistsSync.mockReturnValue(true);
+      mockReadFileSync.mockReturnValue(secretValue);
+      expect(findConfig(key, 'default value')).toEqual(secretValue);
+      expect(mockExistsSync).toHaveBeenCalledWith(`/run/secrets/${key}`);
+      expect(mockReadFileSync).toHaveBeenCalledWith(
+        `/run/secrets/${key}`,
+        'utf-8'
+      );
+      clearConfig(key);
```

Comment:
Would an `afterEach` block be useful here to DRY up the test code?

Reply:
In test code, readability and atomicity are frequently preferred over reducing duplication. `afterEach` is ideal for situations where a testing utility needs to hook some cleanup code after a test has run. Occasionally a group of examples may need similar setup and teardown, and if that group of examples are thought of as a unit (that is, they would change or be removed together) it may be appropriate to reduce duplication if there are many identical lines of code.

In this circumstance, the test writer is trying to make it clear to the reader that the memoized value needs to be cleared for each example to be considered complete, which would be obfuscated by the use of `afterEach`. (And, there is a whole other discussion about why the module memoizes itself instead of there being an intermediate layer.)